react/week4/github-user-search/src/Components/UserSearch.js

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from "react";
+import userSearchContext from "../Context/userSearchContext";
+import Loading from "./Loading";
+
+function UserSearch() {
+  const [name, setName] = useState("");
+  const [data, setData] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  // const [query, setQuery] = useState('')
+
+  // function handleChange(e) {
+  //     e.preventDefault();
+
+  //     setName(e.target.value);
+  // }
+
+  const bucket = {
+    isLoading,
+    setIsLoading,
+  };
+
+  useEffect(() => {
+    { name.length === 0 ? setData(["no result"]) : setIsLoading(true);
```

Comment:
Here you have to extract this line out of useEffect: `{ name.length === 0 ? setData(["no result"]) : setIsLoading(true);`. Inside the useEffet we just handle side effect like fetching, posting etc...
You can  write the logic inside your return statement: `{name.length === 0 ? (<div>No result</div>) : (<div>{data.map((user, index) => (<div key={index}><Names userName={user.login} /></div>))}</div>)}<Loading /></div>`. This is to give you an exemple of what  you can do, there more you have to add.

Reply:
Thanks @Ekamba , I'll implement the changes and merge :)